### PR TITLE
fix: prevent extra half shift

### PR DESCRIPTION
**Pull request by Timbobeek**
This code prevents a user from adding a half-shift if this user already has a half-shift running. Appropriate error message pops up below Employee if attempted.

### Before

[Screencast from 04-28-2024 06:56:29 PM.webm](https://github.com/joshuawootonn/time-track/assets/68671029/607fce86-564e-46e0-96ea-eb3824ab3bdc)

### After

[Screencast from 04-28-2024 06:54:52 PM.webm](https://github.com/joshuawootonn/time-track/assets/68671029/e1f2a9c8-0993-4a8c-976a-a6c48ae517e1)

### CODE REVIEW

File: desktop/src/store/Shift/endpoints.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ export const put = (shift) => {
 
 export const getCurrentShift = (employeeId) => {
   return axios.get(
-    `/employees/${employeeId}/shifts?filter[limit]=1&filter[order]=id DESC`,
+    `/employees/${employeeId}/shifts?filter[limit]=1&filter[order]=clock_out_date`
```
Comment (joshuawootonn) [inline desktop/src/store/Shift/endpoints.js:25]: I would find all references to this and test that those operations still work just to make sure.

**Review by joshuawootonn — APPROVED** [review]
Beautiful

File: desktop/src/store/Shift/endpoints.js (hunk 1 above)
Comment (joshuawootonn) [reply desktop/src/store/Shift/endpoints.js:25]: You could rename this since `current` is vague.

Comment (joshuawootonn) [reply desktop/src/store/Shift/endpoints.js:25]: Maybe `getHalfShiftOrNull`.

Comment (Timbobeek) [reply desktop/src/store/Shift/endpoints.js:25]: Done, renamed all instances!

**Pull request merged**